File: modules/core/misc/java/test/TermCriteriaTest.java
package org.opencv.test.core;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.opencv.core.TermCriteria;
import org.opencv.test.OpenCVTestCase;

public class TermCriteriaTest extends OpenCVTestCase {

    private TermCriteria tc1;
    private TermCriteria tc2;

    @Override
    public void setUp() {
        super.setUp();

        tc1 = new TermCriteria();
        tc2 = new TermCriteria(2, 4, EPS);
    }

    @Test
    public void testClone() {
        tc1 = tc2.clone();
        assertEquals(tc2, tc1);
    }

    @Test
    public void testEqualsObject() {
        assertFalse(tc2.equals(tc1));

        tc1 = tc2.clone();
        assertTrue(tc2.equals(tc1));
    }

    @Test
    public void testHashCode() {
        assertEquals(tc2.hashCode(), tc2.hashCode());
    }

    @Test
    public void testSet() {
        double[] vals1 = {};
        tc1.set(vals1);

        assertEquals(0, tc1.type);
        assertEquals(0, tc1.maxCount);
        assertEquals(0.0, tc1.epsilon, 0);

        double[] vals2 = { 9, 8, 0.002 };
        tc2.set(vals2);

        assertEquals(9, tc2.type);
        assertEquals(8, tc2.maxCount);
        assertEquals(0.002, tc2.epsilon, 0);
    }

    @Test
    public void testTermCriteria() {
        tc1 = new TermCriteria();

        assertNotNull(tc1);
        assertEquals(0, tc1.type);
        assertEquals(0, tc1.maxCount);
        assertEquals(0.0, tc1.epsilon, 0);
    }

    @Test
    public void testTermCriteriaDoubleArray() {
        double[] vals = { 3, 2, 0.007 };
        tc1 = new TermCriteria(vals);

        assertEquals(3, tc1.type);
        assertEquals(2, tc1.maxCount);
        assertEquals(0.007, tc1.epsilon, 0);
    }

    @Test
    public void testTermCriteriaIntIntDouble() {
        tc1 = new TermCriteria(2, 4, EPS);

        assertNotNull(tc2);
        assertEquals(2, tc2.type);
        assertEquals(4, tc2.maxCount);
        assertEquals(EPS, tc2.epsilon, 0);
    }

    @Test
    public void testToString() {
        String actual = tc2.toString();
        double eps = EPS;
        String expected = "{ type: 2, maxCount: 4, epsilon: " + eps + "}";

        assertEquals(expected, actual);
    }

}

File: modules/core/misc/objc/test/TermCriteriaTest.swift
//
//  TermCriteriaTest.swift
//
//  Created by Giles Payne on 2020/01/31.
//

import XCTest
import OpenCV

class TermCriteriaTest: OpenCVTestCase {

    let tc2 = TermCriteria(type: 2, maxCount: 4, epsilon: EPS)

    func testClone() {
        let tc1 = tc2.clone()
        XCTAssertEqual(tc2, tc1)
    }

    func testEqualsObject() {
        var tc1 = TermCriteria()
        XCTAssertFalse(tc2 == tc1)

        tc1 = tc2.clone()
        XCTAssert(tc2 == tc1)
    }

    func testHashCode() {
        XCTAssertEqual(tc2.hash(), tc2.hash())
    }

    func testSet() {
        let tc1 = TermCriteria()
        let vals1:[Double] = []

        tc1.set(vals: vals1 as [NSNumber])

        XCTAssertEqual(0, tc1.type)
        XCTAssertEqual(0, tc1.maxCount)
        XCTAssertEqual(0.0, tc1.epsilon)

        let vals2 = [9, 8, 0.002]
        tc2.set(vals: vals2 as [NSNumber])

        XCTAssertEqual(9, tc2.type)
        XCTAssertEqual(8, tc2.maxCount)
        XCTAssertEqual(0.002, tc2.epsilon)
    }

    func testTermCriteria() {
        let tc1 = TermCriteria()

        XCTAssertNotNil(tc1)
        XCTAssertEqual(0, tc1.type)
        XCTAssertEqual(0, tc1.maxCount)
        XCTAssertEqual(0.0, tc1.epsilon)
    }

    func testTermCriteriaDoubleArray() {
        let vals = [ 3, 2, 0.007]
        let tc1 = TermCriteria(vals: vals as [NSNumber])

        XCTAssertEqual(3, tc1.type)
        XCTAssertEqual(2, tc1.maxCount)
        XCTAssertEqual(0.007, tc1.epsilon)
    }

    func testTermCriteriaIntIntDouble() {
        let tc1 = TermCriteria(type: 2, maxCount: 4, epsilon: OpenCVTestCase.EPS)

        XCTAssertNotNil(tc1)
        XCTAssertEqual(2, tc1.type)
        XCTAssertEqual(4, tc1.maxCount)
        XCTAssertEqual(OpenCVTestCase.EPS, tc1.epsilon)
    }

    func testToString() {
        let actual = "\(tc2)"
        let expected = "TermCriteria { type: 2, maxCount: 4, epsilon: 0.001000}"
        XCTAssertEqual(expected, actual)
    }

}
